frontend/lib/src/components/elements/DeckGlJsonChart/DeckGlJsonChart.tsx

```diff
@@ -170,16 +176,20 @@ export class DeckGlJsonChart extends PureComponent<PropsWithHeight, State> {
 
     const currFullScreen = isFullScreen ?? false
 
-    // Only parse JSON when not transitioning to/from fullscreen, the element id changes, or theme changes
+    // Only parse JSON when not transitioning to/from fullscreen, the element hash changes, or theme changes
     if (
-      element.id !== state.id ||
+      element.hash !== state.hash ||
```

Comment:
Do we even need a backend-computed hash here? A `useEffect` / `useMemo` with `element.json` as a dependency might be sufficient.

Reply:
Oh, that's a great point! I hadn't considered that since I noticed the backend was already hashing this data. Let me do do a revision where we keep it frontend-only.

Reply:
The historical reason why it wasn't considered might have been that it is not as obvious on how to do that with a class-based component.

Reply:
Btw. in the context of the new map features, it might also be an option to directly migrate to functional. This is a bit of a long term goal to have all components migrated to functional.

Reply:
@lukasmasuch I appreciate that context! I've gone ahead and updated this all to leverage modern React in the frontend, which in turn allows this to not utilize a backend hash strategy at all, nice!

Reply:
Awesome 🎉